### PR TITLE
Update ResiReg User Stories on Developer Guide

### PR DESCRIPTION
**Pull request by JingYenLoh**
This PR adds the User Stories our team has discussed to the Developer Guide. We follow the original layout specified in AddressBook.

Addresses part of #51.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -249,14 +249,29 @@ _{Explain here how the data archiving feature will be implemented}_
...
-| `* * *`  | user                                       | find a person by name          | locate details of persons without having to go through the entire list |
-| `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |
-| `*`      | user with many persons in the address book | sort persons by name           | locate a person easily                                                 |
+| Priority | As a…                    | I can…                                                   | So that I can…                                                                                          |
+| -------- | ------------------------ | -------------------------------------------------------- | ------------------------------------------------------------------------------------------------------- |
+| * *      | Meticulous OHS admin     | have automatic backups of my data                        | rest knowing my data will not be accidentally erased.                                                   |
```
Comment (mkeoliya) [inline docs/DeveloperGuide.md]: [These issues](https://github.com/AY2021S1-CS2103-T16-3/tp/issues?page=2&q=is%3Aissue+is%3Aopen+label%3Atype.Story) are missing.

Comment (mkeoliya) [reply docs/DeveloperGuide.md]: Also missing are #27, #25, #24, #23.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -249,14 +249,29 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| * * *    | First Time User          | ask for help                                             | quickly and easily learn how to use the application in one place.                                       |
+| * * *    | OHS Admin                | view a list of all students                              | check which students are in the system and access their particulars.                                    |
+| * * *    | OHS Admin                | so that I can check which students stay in which rooms.  | check which students are in the system and access their particulars.                                    |
+| * *      | As as skeptical GUI user | create aliases to other commands                         | perform my common actions while typing less.                                                            |
+| * *      | skeptical GUI user       | create compound commands                                 | ... e.g. to view bills and pay them at the same time, perform common sequences of actions more quickly. |
```
Comment (mkeoliya) [inline docs/DeveloperGuide.md]: Should be removed.

**Review by mkeoliya — COMMENTED** [review]
A few user stories are missing, but it's a-ok besides that.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (JingYenLoh) [reply docs/DeveloperGuide.md]: I'll remove the redundant line. The rest shouldn't be removed IMO - they are stories we've come up with.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (JingYenLoh) [reply docs/DeveloperGuide.md]: #27 is already in. I'll include the rest.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (mkeoliya) [reply docs/DeveloperGuide.md]: Yep my bad! Was referring to the `create compound commands` user story only. Could you remove it?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (mkeoliya) [reply docs/DeveloperGuide.md]: Perhaps you forgot to include #58 and #59?

**Review by mkeoliya — APPROVED** [review]
LGTM!

**Pull request merged**